xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -123,14 +128,44 @@ bool CVAAPIContext::EnsureContext(CVAAPIContext **ctx, CDecoder *decoder)
   return true;
 }
 
+namespace
+{
+int find_open_render_node_fd()
```

Comment:
You seem not to understand. Are you saying the folks who do vaapi are wrong in providing backend code? You have to find the correct drm fd for the display in use. Pulling this code from vaapi, where it belongs to, to the application level is wrong.